test/e2e/framework/node/resource.go

```diff
@@ -156,7 +156,6 @@ func IsConditionUnset(node *v1.Node, conditionType v1.NodeConditionType) bool {
 
 // Filter filters nodes in NodeList in place, removing nodes that do not
 // satisfy the given condition
-// TODO: consider merging with pkg/client/cache.NodeLister
```

Comment:
removing this TODO is fine. the framework should not use code from /pkg.

where is this function used?

i see these:
https://github.com/kubernetes/kubernetes/blob/a18303dfa4a24dcb81f061b7b959f870180bb5cd/test/e2e/cloud/gcp/ha_master.go#L133-L147

anywhere else?

the name "Filter" is quite generic and the function should be either moved or renamed.

Reply:
I see there
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/network_partition.go#L149
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/network_partition.go#L498
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/cluster_size_autoscaling.go#L1358
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/cloud/gcp/ha_master.go#L133
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/cloud/gcp/ha_master.go#L145

in package:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L190
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L345
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/node/resource.go#L394

it is not easy to move  due to packages reference each other.rename it looks ok.

Reply:
Filter -> FilterNode?

Reply:
let's leave it named Filter for now.

Reply:
done